packages/snaps-controllers/src/interface/SnapInterfaceController.test.tsx

```diff
@@ -29,13 +37,82 @@ jest.mock('@metamask/snaps-utils', () => ({
 }));
 
 describe('SnapInterfaceController', () => {
+  it('handles a notificationsListUpdated event on the controller messenger', async () => {
```

Comment:
Can we expand this test to test that it only removes interfaces from notifications that were removed from the list? Right now we aren't properly testing the case where we want to remove 1 of 2 or more notifications.